### PR TITLE
Rename `deprovision` to `ban` endpoint

### PR DESCRIPTION
**Pull request by xcoulon**
Create a new `ban` endpoint and internally
forward all incoming requests from `deprovision`
to `ban`
Also, rename all references of `deprovision` to `ban`,
including a new column `ban` in the database, containing
the same values as its sibling column `deprovision`.
Both columns should be updated in parallel until the transition
to the new endpoint is complete.

Fixes #ODC117

### CODE REVIEW

File: authentication/account/service/user.go [hunk 1]
```diff
@@ -31,9 +31,9 @@ type userServiceImpl struct {
 	base.BaseService
 }
 
-// ResetDeprovisioned sets User.Deprovisioned to false
+// ResetBanned sets User.Banned to false
 func (s *userServiceImpl) ResetDeprovision(ctx context.Context, user repository.User) error {
```
Comment (alexeykazakov) [inline authentication/account/service/user.go]: You changed the comment but the method is still ResetDeprovision.

Comment (xcoulon) [reply authentication/account/service/user.go]: ah right! Let me fix that ;)

Comment (xcoulon) [reply authentication/account/service/user.go]: addressed in f1fff0e

**Comment by alexeykazakov** [comment]
‘’’
ERROR: These files differ from gofmt's style (run 'make format-go-code' to fix this):
./test/account.go
‘’’

**Comment by xcoulon** [comment]
> ‘’’
> ERROR: These files differ from gofmt's style (run 'make format-go-code' to fix this):
> ./test/account.go
> ‘’’

ahem, let me fix that 😬

**Comment by xcoulon** [comment]
> > ‘’’
> > ERROR: These files differ from gofmt's style (run 'make format-go-code' to fix this):
> > ./test/account.go
> > ‘’’
> 
> ahem, let me fix that 😬

fixed in 2d3f897

**Pull request merged**